server/src/instant/db/model/attr.clj

```diff
@@ -681,59 +656,82 @@
 ;; Creates a wrapper over attrs. Makes them act like a regular list, but
 ;; we can also index them on demand so that our access patterns will be
 ;; efficient.
-(deftype Attrs [elements cache]
+(declare wrap-attrs)
+
+(deftype Attrs [elements
+                by-id-cache
+                by-fwd-ident-cache
+                by-rev-ident-cache
+                ids-by-etype-cache
+                ea-ids-by-etype-cache]
   clojure.lang.ISeq
   (count [_this]
     (count elements))
   (first [_this]
     (first elements))
   (next [_this]
-    (let [nxt (next elements)]
-      (if nxt
-        (Attrs. nxt (delay (index-attrs nxt)))
-        nil)))
+    (some-> elements next wrap-attrs))
   (more [_this]
-    (if-let [nxt (next elements)]
-      (Attrs. nxt (delay (index-attrs nxt)))
+    (if-some [nxt (next elements)]
+      (wrap-attrs nxt)
       clojure.lang.PersistentList/EMPTY))
   (empty [_this]
-    (Attrs. () (delay {})))
+    (wrap-attrs ()))
   (equiv [_this other]
     (= elements other))
   (cons [_this o]
-    (let [new-elements (cons o elements)]
-      (Attrs. new-elements (delay (index-attrs new-elements)))))
+    (wrap-attrs (cons o elements)))
   (seq [this]
     (if (empty? elements)
       nil
       this))
 
+  clojure.lang.IHashEq
```

Comment:
Oops, I thought iseq handled the hash for me--we've been sending unnecessary spam whenever the attrs got pushed out of the cache all this time!